provisioning/hare_setup.py

```diff
@@ -0,0 +1,279 @@
+        try:
+            if not is_cluster_running() and bootstrap_cluster() != 0:
+                logging.error('Failed to bootstrap the custer')
+                exit(1)
+
+            while is_cluster_running():
+                shutdown_cluster()
+
+        except Exception as error:
+            logging.error('Error while initializing the cluster (%s)',
+                          error)
+            exit(-1)
+
+class Test(argparse.Action):
+    """
+    Read CDF file and compare fields with output of 'hctl status'
+    Will start the cluster(and shutdown before exiting) if not already started
+    """
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            if not is_cluster_running() and bootstrap_cluster() != 0:
+                logging.error('Failed to bootstrap the cluster')
+                exit(-1)
+
+            cluster_status = check_cluster_status()
+
+            while is_cluster_running():
+                shutdown_cluster()
+
+            if cluster_status:
+                logging.error('Cluster status reports failure')
+                exit(-1)
+
+        except Exception as error:
+            logging.error('Error while checking cluster status (%s)',
+                          error)
+            exit(-1)
+
+class SupportBundle(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            # Default target directory is /tmp/hare
+            if os.system('hctl reportbug') != 0:
+                logging.error('Failed to generate support bundle')
+                exit(-1)
+        except Exception as error:
+            logging.error('Error while generating support bundle (%s)',
+                          error)
+            exit(-1)
+
+def checkRpm(rpm_name):
+    return os.system('rpm -qa | grep -q ' + rpm_name)
+
+
+def is_cluster_running() -> bool:
+    if os.system('hctl status >/dev/null') == 0:
+        return True
+    else:
+        return False
+
+
+def bootstrap_cluster():
+    return os.system('hctl bootstrap --mkfs /var/lib/hare/cluster.yaml')
+
+
+def shutdown_cluster():
+    return os.system('hctl shutdown')
+
+
+def list2dict(nodes_data_hctl: list) -> dict:
+    node_info_dict = {}
+    for node in nodes_data_hctl:
+        node_svc_info = {}
+        for service in node['svcs']:
+            node_svc_info[service['name']] = []
+            if (service['status'] == 'started'):
+                node_svc_info[service['name']].append(service['status'])
+        node_info_dict[node['name']] = node_svc_info
+
+    return node_info_dict
+
+
+def check_cluster_status():
+    cluster_desc = None
+    with open("/var/lib/hare/cluster.yaml", 'r') as stream:
+        cluster_desc = yaml.safe_load(stream)
+    cmd = ['hctl', 'status', '--json']
+    cluster_info_out, err = execute_command_and_get_output(cmd)
+    if err:
+        logging.error('Failed to get hctl status')
+        return -1
```

Comment:
What is the point in returning a magic number? If there is a problem - please raise an exception.

Reply:
Done